python/plugins/processing/gui/Postprocessing.py

```diff
@@ -36,18 +36,41 @@
                        QgsProcessingUtils,
                        QgsMapLayer,
                        QgsWkbTypes,
-                       QgsMessageLog)
+                       QgsMessageLog,
+                       QgsProviderRegistry)
 
 from processing.core.ProcessingConfig import ProcessingConfig
 from processing.gui.RenderingStyles import RenderingStyles
 
 
+def set_layer_name(layer, context_layer_details):
+    """
+    Sets the name for the given layer, either using the layer's file name
+    (or database layer name), or the name specified by the parameter definition.
+    """
+    use_filename_as_layer_name = ProcessingConfig.getSetting(ProcessingConfig.USE_FILENAME_AS_LAYER_NAME)
+
+    if use_filename_as_layer_name or not context_layer_details.name:
+        source_parts = QgsProviderRegistry.instance().decodeUri(layer.dataProvider().name(), layer.source())
+        layer_name = source_parts.get('layerName', '')
+        # if source layer name exists, use that -- else use
+        if layer_name:
+            layer.setName(layer_name)
+        else:
+            path = source_parts.get('path', '')
+            if path:
```

Comment:
is it unsafe to leave without else condition? in that case layer.setName will be not called at all.

Reply:
Should we just fallback to the output name in this case?

Reply:
ha ok, outputName is already set because it is a postProcesing step? for me it's ok to use ouput name.